lib/membrane_rtc_engine/engine.ex

```diff
@@ -1094,13 +1151,37 @@ defmodule Membrane.RTC.Engine do
     |> Enum.filter(&Map.has_key?(ctx.children, &1))
   end
 
-  defp do_publish(_endpoints, _ctx, {_, []} = _tracks, _endpoint_bin), do: []
+  defp do_publish(_endpoints, _ctx, {_, []} = _tracks, _endpoint_bin, _state), do: []
+
+  defp do_publish(endpoints, ctx, {:new_tracks, _tracks} = msg, endpoint_bin_name, _state) do
+    flat_map_children(ctx, fn
+      {:endpoint, endpoint_id} = other_endpoint_bin ->
+        endpoint = Map.get(endpoints, endpoint_id)
+
+        if other_endpoint_bin != endpoint_bin_name and not is_nil(endpoint) do
+          [forward: {other_endpoint_bin, msg}]
+        else
+          []
+        end
+
+      _child ->
+        []
+    end)
+  end
 
-  defp do_publish(endpoints, ctx, msg, endpoint_bin_name) do
+  defp do_publish(endpoints, ctx, {type, tracks}, endpoint_bin_name, state) do
     flat_map_children(ctx, fn
       {:endpoint, endpoint_id} = other_endpoint_bin ->
         endpoint = Map.get(endpoints, endpoint_id)
 
+        has_subscription_on_track = fn track_id ->
+          subscribed_tracks = Map.get(state.subscriptions, endpoint_id)
+          subscribed_tracks != nil and Map.has_key?(subscribed_tracks, track_id)
```

Comment:
I think we should add entry in `state.subscription` for each newly added endpoint. Then you can write
```suggestion
          state.subscriptions
          |> Map.fetch!(endpoint_id)
          |> Map.has_key?(track_id)
```